axlearn/experiments/run_aot_compilation.py

```diff
@@ -0,0 +1,78 @@
+# Copyright © 2023 Apple Inc.
+"""A command-line tool to perform AoT (ahead-of-time) compilation.
+
+pip install 'jax[tpu]==0.4.21' -f https://storage.googleapis.com/jax-releases/libtpu_releases.html
+python axlearn/experiments/run_aot_compilation.py \
+    --config_module=text.gpt.c4_trainer \
+    --config=fuji-7B \
+    --topology=v4-1024 1> /tmp/aot_stdout 2| tee /tmp/aot_stderr
+
+Reference:
+https://docs.google.com/document/d/1Y5IdmvAZA7UtMHAWkRh8k2PscVoG5FvMH9-E6hygsyY/
+"""
+
+import pickle
+from typing import Optional
+
+from absl import flags, app
+from jax.experimental.serialize_executable import serialize
+
+from axlearn.common.aot_compilation import compile_trainer_programs
+from axlearn.common.trainer import SpmdTrainer
+from axlearn.common.utils import set_data_dir
+from axlearn.common.utils_spmd import setup
+from axlearn.experiments import TrainerConfigFn, get_named_trainer_config
+
+
+flags.DEFINE_string("config_module", None, "The TPU topology.")
+flags.DEFINE_string("config", None, "The TPU topology.")
+flags.DEFINE_string("topology", None, "The TPU topology.")
+flags.DEFINE_integer("topology_num_slices", 1, "The number of TPU slices.")
+
+FLAGS = flags.FLAGS
+
+
+def _compile_and_dump_programs(
```

Comment:
Would it make sense to have this be in `aot_compilation`? E.g. we seem to do the same thing in aot_test.

Reply:
Removed duplicate code from aot_test.